### ISSUE TITLE
Integration test for runtime upgrades

### ISSUE THREAD

**Issue by bkchr** [paritytech/cumulus#360]
We need an integration test for runtime upgrades.

1. Use `schedule_upgrade` to schedule the upgrade. This will require that we bump the `spec_version`. We will need to add an `if` to this function https://github.com/paritytech/cumulus/blob/bkchr-update-yea-yeah-yeah/test/runtime/src/lib.rs#L298. Basically we want to check the storage for some value to be set (this value needs to be set at genesis). If this value is set, we return `VERSION` and otherwise we return the "custom" version where `spec_version` is bumped. This will make it possible to make the checks happy in `schedule_upgrade` without having to build the runtime twice.
2. There should a custom `OnRuntimeUpgrade` code for the runtime that sets some value which we will query to ensure that the upgrade actually happened.
3.
4. Profit

TLDR: Add a new integration test that schedules a runtime upgrade and we check that this runtime upgrade was executed.
